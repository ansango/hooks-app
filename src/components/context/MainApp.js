export const MainApp = () => {
  return <div>Context</div>;
};
