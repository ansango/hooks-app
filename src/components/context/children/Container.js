
import "./Container.css";
export const Container = ({ title = "", children }) => {
  return (
    <div className="base-container py-5 bg-light">
      <h3 className="text-center">{title}</h3>
      <div className="px-2">{children}</div>
    </div>
  );
};
