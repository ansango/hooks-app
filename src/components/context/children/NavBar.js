import { NavLink } from "react-router-dom";

export const NavBar = ({ routes }) => {
  return (
    <nav className="navbar navbar-light bg-light d-flex flex-row justify-content-end">
      <div className="navbar-nav">
        <div className="d-flex">
          {routes.map((route) => (
            <NavLink
              key={route.name}
              exact
              activeClassName="active"
              className="nav-item nav-link mx-3"
              to={route.path}
            >
              {route.name}
            </NavLink>
          ))}
        </div>
      </div>
    </nav>
  );
};
