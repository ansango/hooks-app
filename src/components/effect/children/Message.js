import { useEffect, useState } from "react";

export const Message = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 });
  const { x, y } = coords;
  useEffect(() => {
    const mouseMove = (e) => setCoords({ x: e.x, y: e.y });
    window.addEventListener("mousemove", mouseMove);
    return () => {
      window.removeEventListener("mousemove", mouseMove);
    };
  }, []);
  return (
    <div>
      X: <p>{x}</p>
      Y: <p>{y}</p>
    </div>
  );
};
