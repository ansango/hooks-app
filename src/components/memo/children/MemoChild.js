import React from "react";

export const MemoChild = React.memo(({ number, increment }) => {
  console.log("render");

  return (
    <button className="btn btn-primary me-3" onClick={() => increment(number)}>
      {number}
    </button>
  );
});
