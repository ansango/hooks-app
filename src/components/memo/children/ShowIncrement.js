import React from "react";
import { log } from "../../../helpers/log";

export const ShowIncrement = React.memo(({ increment }) => {
  log("ShowIncrement", "Render");
  return (
    <button className="btn btn-primary" onClick={() => increment(5)}>
      Incrementar
    </button>
  );
});
