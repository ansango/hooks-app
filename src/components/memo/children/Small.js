import React from "react";
import { log } from "../../../helpers/log";

export const Small = React.memo(({ value }) => {
  log("Small", "Render");
  return <small>{value}</small>;
});
