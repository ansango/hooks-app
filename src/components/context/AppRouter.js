import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { NavBar } from "./children/NavBar";
import { LogoutPage } from "./children/pages/LogoutPage";
import { HomePage } from "./children/pages/HomePage";
import { LoginPage } from "./children/pages/LoginPage";

const routes = [
  { path: "/", component: HomePage, name: "Home" },
  { path: "/logout", component: LogoutPage, name: "Logout" },
  { path: "/login", component: LoginPage, name: "Login" },
];

export const AppRouter = () => {
  return (
    <Router>
      <NavBar routes={routes} />

      <Switch>
        {routes.map((route) => (
          <Route
            exact
            key={route.name}
            path={route.path}
            component={route.component}
          />
        ))}
        <Redirect to="/" />
      </Switch>
    </Router>
  );
};
