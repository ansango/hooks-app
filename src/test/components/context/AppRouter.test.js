import { AppRouter } from "../../../components/context/AppRouter";
import { UserContext } from "../../../components/context/children/UserContext";
import { mount } from "enzyme";

describe("Test in <AppRouter/>", () => {
  const user = { id: 123, name: "Anibal" };
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
