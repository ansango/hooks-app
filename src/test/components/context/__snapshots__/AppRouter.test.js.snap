// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter/> should render 1`] = `
Array [
  <nav
    className="navbar navbar-light bg-light d-flex flex-row justify-content-end"
  >
    <div
      className="navbar-nav"
    >
      <div
        className="d-flex"
      >
        <a
          aria-current="page"
          className="nav-item nav-link mx-3 active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link mx-3"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link mx-3"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="base-container py-5 bg-light"
  >
    <h3
      className="text-center"
    >
      Home
    </h3>
    <div
      className="d-flex flex-column align-items-center justify-content-center p-5"
    >
      <pre>
        {
   "id": 123,
   "name": "Anibal"
}
      </pre>
    </div>
  </div>,
]
`;
